tests/upgrades/test_syncplan.py

```diff
@@ -56,6 +56,49 @@ def test_pre_sync_plan_migration(self, request):
         product = product.read()
         assert product.sync_plan.id == sync_plan.id
 
+    @pre_upgrade
+    def test_pre_disabled_sync_plan_logic(self, request):
+        """Pre-upgrade scenario that creates a sync plan with both disabled and enabled recurring logic.
+
+        :id: c75bd43d-d868-461a-9fc3-a1fc7dccc77a
+
+        :steps:
+            1. Create Product
+            2. Create Sync Plan
+            3. Assign sync plan to product
+            4. Disable the sync plan's recurring logic
+            5. Re enable the sync plan
+
+        :expectedresults: Sync plan is created and assigned to a product. The associated recurring
+        logic is cancelled and then the plan is re-enabled so that it gets a new recurring logic.
+
+        :BZ: 1887511
+
+        """
+        count = 0
+        org = entities.Organization(name=f'{request.node.name}_org').create()
+        sync_plan = entities.SyncPlan(
+            organization=org, name=f'{request.node.name}_syncplan', interval="weekly"
+        ).create()
+        product = entities.Product(organization=org, name=f'{request.node.name}_prod').create()
+        entities.Repository(product=product, name=f'{request.node.name}_repos').create()
+        sync_plan.add_products(data={'product_ids': [product.id]})
+        product.sync()
+        product = product.read()
+        assert product.sync_plan.id == sync_plan.id
+        # Note the recurring logic ID for later assert a new one was created
+        count = sync_plan.foreman_tasks_recurring_logic.id
+        # Cancel the recurring logic
+        entities.RecurringLogic(id=count).read()
+        entities.RecurringLogic(id=count).cancel()
+        # Re-enable the sync plan (it will get a new recurring logic)
+        sync_plan.enabled = True
+        sync_plan.update(['enabled'])
+        sync_plan = sync_plan.read()
+        assert sync_plan.enabled
+        # Assert a new recurring logic was assigned
+        assert sync_plan.foreman_tasks_recurring_logic.id > count
```

Comment:
why rely on the fact that IDs represent count, doing just new_id != old_id would be more change-proof

Reply:
I'd agree here, calling it `count` is misleading.

Reply:
I guess I chose "count" as the sync plan ID is incremented, will change it as suggested

Reply:
As this was already merged, I created new PR https://github.com/SatelliteQE/robottelo/pull/9125